### DISCUSSION TITLE
Do you always need to opt-in to get `configMigration` PRs?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#16352] [Request Help] [answered]
1. Do you always need to opt-in for config migration PRs by setting `configMigration=true`, no matter _how_ you run Renovate?
2. Does the GitHub hosted app raise config migration PRs if needed?
3. Do we intend for config migration to be hidden from our user by default?

Reading the docs [`configMigration`](https://docs.renovatebot.com/configuration-options/#configmigration) defaults to `false`. Our [`config:base`](https://docs.renovatebot.com/presets-config/#configbase) preset doesn't mention `configMigration`.

**Comment by rarkins** [comment, marked as answer]
1. Yes
2. No
3. It's not hidden - you can see it in the debug logs and docs

This will create a huge number of PRs if we enable it by default. It needs more testing (including our own) before we're sure it's bug-free and good enough. Ideally also all migration refactors would be completed.

**Reply by HonkingGoose** [reply]
Thank you for the quick response! ❤️ 

> This will create a huge number of PRs if we enable it by default. It needs more testing (including our own) before we're sure it's bug-free and good enough. Ideally also all migration refactors would be completed.

My idea was to enable `configMigration` by default eventually, so the users' `renovate.json` always uses current terms. This means you can always find your config options in the documentation, which prevents confusion.

Check out this query on GitHub: [GitHub code search for `masterIssue` in `renovate.json` files](https://github.com/search?q=masterIssue+filename%3Arenovate.json+language%3AJSON+language%3AJSON&type=Code&ref=advsearch&l=JSON&l=JSON).

Do you want a new feature request issue from me to track making `configMigration=true` the default behavior?

**Reply by rarkins** [reply]
Yes, once I merge #16355 please then add `experimentalIssues` linking to a new issue you create called something like "Enable config migration by default"

**Reply by HonkingGoose** [reply]
See:

- #16359
- #16360
